filesystem/src/main/java/org/frankframework/filesystem/FileSystemListener.java

```diff
@@ -304,28 +317,25 @@ public Message extractMessage(@Nonnull RawMessageWrapper<F> rawMessage, @Nonnull
 		log.debug("Extract message from raw message");
 		try {
 			F file = rawMessage.getRawMessage();
-			if (StringUtils.isEmpty(getMessageType()) || getMessageType().equalsIgnoreCase("name")) {
-				return new Message(getFileSystem().getName(file));
-			}
-			if (StringUtils.isEmpty(getMessageType()) || getMessageType().equalsIgnoreCase("path")) {
-				return new Message(getFileSystem().getCanonicalName(file));
-			}
-			if (getMessageType().equalsIgnoreCase("contents")) {
-				return getFileSystem().readFile(file, getCharset());
-			}
-			if (getMessageType().equalsIgnoreCase("info")) {
-				return new Message(FileSystemUtils.getFileInfo(getFileSystem(), file, getOutputFormat()));
-			}
 
-			Map<String,Object> attributes = getFileSystem().getAdditionalFileProperties(file);
-			if (attributes != null) {
-				Object result = attributes.get(getMessageType());
-				if (result != null) {
-					return Message.asMessage(result);
+			return switch (getMessageType().name()) {
+				case "NAME" -> new Message(getFileSystem().getName(file));
+				case "PATH" -> new Message(getFileSystem().getCanonicalName(file));
+				case "CONTENTS" -> getFileSystem().readFile(file, getCharset());
+				case "INFO" -> new Message(FileSystemUtils.getFileInfo(getFileSystem(), file, getOutputFormat()));
+				case "ATTRIBUTE", "HEADER" -> {
+					Map<String,Object> attributes = getFileSystem().getAdditionalFileProperties(file);
+					if (attributes != null) {
+						Object result = attributes.get(getSearchKey());
+						if (result != null) {
+							yield Message.asMessage(result);
+						}
+					}
+					log.warn("no attribute [{}] found for file [{}]", getMessageType(), getFileSystem().getName(file));
+					yield null;
```

Comment:
Dit mag er uit hoor. Slaat nergens op om een attribute ergens uit te halen dat in `INFO` zit. Dan moet men maar een xpath oid doen.

Reply:
Ik zie dat je dan wel info in het mail fs moet zetten. Ik weet niet of daar al tests voor zijn. Ik ga er van uit dat `FileSystemUtils.getFileInfo` ook gewoon werkt voor het mailfilesystem. (En daar zitten de 'additionalProperties' in.)

Reply:
De releases.md heb ik in https://github.com/frankframework/frankframework/pull/7290 aangemaakt. Maar dat is nog niet gemerged (dus je hebt er nog niet veel aan).

Reply:
Dus nog eventjes wachten tot jouw issue op master staat